main.go

```diff
@@ -476,7 +476,9 @@ func processLogFile(file string, enc *json.Encoder) (int, error) {
 	offset := state.Offset
 	// Are we looking at freshly rotated file since last time we run?
 	// If so let's reset the offset back to 0 and read the file again
-
+	if offset < 0 {
+		offset = 0
+	}
```

Comment:
Under what circumstances does this condition occur? Seems surprising.

Reply:
It shouldn't occur.  But if it does I want to catch it. Should I error here instead?

Reply:
Hmm I think so, it's basically an assertion failure, may as well panic.

Reply:
okay ill change it as a non zero exit with a message

Reply:
latest commits change the condition to an error, and adds a test for the error condition